src/engines.jl

```diff
@@ -32,96 +23,34 @@ function _wait_till_provisioned(engine_name, max_wait_time_s=600)
 end
 
 """
-    create_default_engine(name::String)
+    create_default_engine(name::String; size::String)
 
-Create an XS engine with default settings and the provided name.
+Create an engine with the default configuration.
 
 If the engine already exists, return immediately. If not, create the engine then
 return once the provisioning process is complete, or failed.
 """
-function create_default_engine(name::String)
-    size = "XS"
+function create_default_engine(name::String; size::String="XS")
     max_wait_time_s = 600
 
     try
-        get_engine(get_context(), name; readtimeout=max_wait_time_s)
-        # The engine already exists so return it immediately
-        return name
-    catch
-        # Engine does not exist yet so we'll need to create it
+        create_engine(get_context(), name; size, readtimeout=max_wait_time_s)
+    catch e
+        # If the status code is 409 then the engine already exists and we can wait for it
+        # to be ready
+        if e.status_code != 409
```

Comment:
it's possible `e` doesn't have a `status_code` which could cause this to fail - should this check the type of `e` as well?
```suggestion
        if !(e isa ErrorTypeHere) || e.status_code != 409
```

Reply:
Good catch! Fixed (with HTTPError)